---

---

## rename
rename 's/要替换的内容/替换成的内容/' file    ###文件重命名，C++语言

rename 需替换内容 替换成的内容 file          ####文件重命名，pel语言

## awk
ls *.gz > fastp.lst

awk '{print "fastp -i " $1" -o clean_data/"$1" -h “$1".html -j "$1" .json &" }' fastp.lst > run_fastp.sh  ###fastp批量运行

awk '{print "fastp -i "$1"-o clean_data/"$1".clean.fq" -I "$2" -O clean_data/"$2".clean.fq -f 15 -F 15" }' samplelist.txt > run_fastptrim.sh  ####fastp批量修剪双端数据


gffread -T -o file.gtf file.gff3    ###gff3文件转换成gtf文件



## conda

conda search package             #####在conda库里面搜索相关软件

conda install fastx tookit      #####安装fastx toolkit

fasta_formatter -i genome.fa -o genome_formatter.fa -w 70    ###将基因组的序列按照每行70个碱基排列

## 数据合并（cat-行合并，paste-列合并）









## 3-7-6genomeCheck

bwa **index** -a bwtsw -p 3-7-6 nextgraph.assembly.contig.polished.fasta

bwa **mem** -t 8 ../genome/3-7-6genome/nextgraph.assembly.contig.polished.fasta 2_5_1Fves_Strawberry_P656-D17_good_1.fq.gz 2_5_1Fves_Strawberry_P656-D17_good_2.fq.gz | samblaster --excludeDups --addMateTags --maxSplitCount 2 --minNonOverlap 20 | samtools view -S -b - > **BWA2to3.bam**

samtools view -b -F 1294 **BWA2to3.bam** | samtools sort - > **BWA2to3.discordants.sorted.bam**

bwa mem -t 8 ../genome/3-7-6genome/nextgraph.assembly.contig.polished.fasta 2_5_1Fves_Strawberry_P656-D17_good_1.fq.gz 2_5_1Fves_Strawberry_P656-D17_good_2.fq.gz |samtools view -bS - |samtools sort - >**bwa2to3sort.bam**

samtools flagstat **bwa2to3sort.bam**

41971158 + 0 in total (QC-passed reads + QC-failed reads)
0 + 0 secondary
233014 + 0 supplementary
0 + 0 duplicates
41263157 + 0 mapped (98.31% : N/A)
41738144 + 0 paired in sequencing
20869072 + 0 read1
20869072 + 0 read2
39195322 + 0 properly paired (93.91% : N/A)
40924396 + 0 with itself and mate mapped
105747 + 0 singletons (0.25% : N/A)
583154 + 0 with mate mapped to a different chr
293945 + 0 with mate mapped to a different chr (mapQ>=5)

samtools view -bs -f2 **bwa2to3sort.bam**

报错：samtools view: Incorrect sampling argument "-f2"

bedtools genomecov -d -split -ibam **bwa2to3sort.bam** >bwa2to3_cov.txt







hisat2-build -p 8 nextgraph.assembly.contig.polished.fasta 3-7-6

hisat2 -p 8 --dta -x /mnt/data/duhaiyuan/SM_RNA_seq/hisat2/siji_index 1.fq -2 2.fq -S filename.sam













